compose.py

```diff
@@ -0,0 +1,305 @@
+ENTRYPOINT ["/opt/tritonserver/nvidia_entrypoint.sh"]
+'''
+    with open(os.path.join(ddir, dockerfile_name), "a") as dfile:
+        dfile.write(df)
+
+### create build container
+def build_docker_container(ddir, dockerfile_name, container_name):
+    # Before attempting to run the new image, make sure any
+    # previous <container_name> container is removed.
+    client = docker.from_env(timeout=3600)
+    try:
+        existing = client.containers.get(container_name)
+        existing.remove(force=True)
+    except docker.errors.NotFound:
+        pass  # ignore
+
+    p = subprocess.Popen(['docker', 'build', '-t', container_name, '-f', \
+        os.path.join(ddir, dockerfile_name), '.'])
+    p.wait()
+    fail_if(p.returncode != 0, 'docker build {} failed'.format(container_name))
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    group_qv = parser.add_mutually_exclusive_group()
+    group_qv.add_argument('-q',
+                          '--quiet',
+                          action="store_true",
+                          required=False,
+                          help='Disable console output.')
+    group_qv.add_argument('-v',
+                          '--verbose',
+                          action="store_true",
+                          required=False,
+                          help='Enable verbose output.')
+    parser.add_argument(
+        '--build-name',
+        type=str,
+        required=False,
+        help=
+        'Build name. Default is "tritonserver_build".'
+    )
+    parser.add_argument(
+        '--build-dir',
+        type=str,
+        required=True,
+        help=
+        'Build directory. All repo clones and builds will be performed in this directory.'
+    )
+    parser.add_argument(
+        '--version',
+        type=str,
+        required=False,
+        help=
+        'The Triton version. If not specified defaults to the value in the TRITON_VERSION file.'
+    )
+    parser.add_argument(
+        '--container-version',
+        type=str,
+        required=False,
+        help=
+        'The Triton container version to build. If not specified the container version will be chosen automatically based on --version value.'
+    )
+    parser.add_argument(
+        '--upstream-container-version',
+        type=str,
+        required=False,
+        help=
+        'The upstream container version to use for the build. If not specified the upstream container version will be chosen automatically based on --version value.'
+    )
+    parser.add_argument(
+        '--backend',
+        action='append',
+        required=False,
+        help=
+        'Include specified backend in build as <backend-name>. To have multiple backends built, specify --backend <foo> --backend <bar>'
+    )
+    parser.add_argument(
+        '--repoagent',
+        action='append',
+        required=False,
+        help='Include specified repo agent in build as <repoagent-name>')
+    FLAGS = parser.parse_args()
+
```

Comment:
Add --enable-gpu as in build.py. For now not supporting cpu-only build so should raise error if --enable-gpu is not specified.